uk.co.compucorp.civicrm.hrcore/gulp/tasks/backstopjs.js

```diff
@@ -0,0 +1,236 @@
+var Promise = require('es6-promise').Promise;
+
+var BACKSTOP_DIR = 'backstop_data/';
+var BACKSTOP_DIR_PATH = path.join(__dirname, '..', '..', BACKSTOP_DIR);
+var DEFAULT_CREDENTIAL = 'admin';
+var FILES = { config: 'site-config.json', tpl: 'backstop.tpl.json' };
+var CONFIG_TPL = {
+  'url': 'http://%{site-host}',
+  'credentials': {
+    'admin': { 'name': '%{admin-name}', 'pass': '%{admin-password}' },
+    'manager': { 'name': '%{manager-name}', 'pass': '%{manager-password}' },
+    'staff': { 'name': '%{staff-name}', 'pass': '%{staff-password}' }
+  }
+};
+
+module.exports = [
+  {
+    name: 'backstopjs:reference',
+    fn: function () {
+      return runBackstopJS('reference');
+    }
+  },
+  {
+    name: 'backstopjs:test',
+    fn: function () {
+      return runBackstopJS('test');
+    }
+  },
+  {
+    name: 'backstopjs:report',
+    fn: function () {
+      return runBackstopJS('openReport');
+    }
+  },
+  {
+    name: 'backstopjs:approve',
+    fn: function () {
+      return runBackstopJS('approve');
+    }
+  }
+];
+
+/**
+ * Constructs URL for BackstopJS scenario based on
+ * site URL, scenario config URL and contact "roles" and IDs map
+ *
+ * @param  {String} siteUrl
+ * @param  {String} scenarioUrl
+ * @param  {Object} contactIdsByRoles
+ * @return {String}
+ */
+function constructBackstopJSScenarioUrl (siteUrl, scenarioUrl, contactIdsByRoles) {
+  scenarioUrl = scenarioUrl.replace(/\{\{contactId:([^}]+)\}\}/g, function (fullMatch, contactRole) {
+    return contactIdsByRoles[contactRole];
+  });
+
+  return siteUrl + '/' + scenarioUrl;
+}
+
+/**
+ * Fetches civicrm contacts who's emails match "civihr_" pattern
+ * and returns a map of their "roles" connected to their IDs.
+ * Requires 'civihr_(staff|manager|admin)@...' to be presented in DB,
+ * otherwise will throw an error.
+ *
+ * @return {Promise} resolved with {Object}, ex. { 'staff': 204, ... etc }
+ */
+function getRolesAndIDs () {
+  return new Promise(function (resolve, reject) {
+    exec('cv api contact.get sequential=1 email="civihr_%" contact_type="Individual" return="email,contact_id"', function (err, result) {
+      var idsByRoles, missingRoles;
+
+      if (err) {
+        return reject(new Error('Unable to fetch contact roles and IDs: ' + err));
+      }
+
+      idsByRoles = _(JSON.parse(result).values)
+        .map(function (contact) {
+          var role = contact.email.split('@')[0].split('_')[1];
+
+          return [role, contact.contact_id];
+        })
+        .fromPairs()
+        .value();
+
+      missingRoles = _.difference(['staff', 'manager', 'admin'], _.keys(idsByRoles));
+
+      if (missingRoles.length) {
+        return reject(new Error('Required users with emails ' + missingRoles.map(function (role) {
+          return 'civihr_' + role + '@*';
+        }).join(', ') + ' were not found in the database'));
+      }
+
+      resolve(idsByRoles);
+    });
+  });
+}
+
+/**
+ * Checks if the site config file is in the backstopjs folder
```

Comment:
The description to this function and the name do not really play well together.

`checkConfigFileAndCreateIfDoesNotExist`
`touchConfigFile`
`findOrCreateConfigFile` - not really since in that case I might expect config file to be returned